Refactor

bhornseth opened the pull request.

bhornseth committed:
Eliminate test warnings

Each of these changes were causing a longer version of the below warning

> WARNING: Using the `raise_error` matcher without providing a specific error
> or message risks false positives

There are definitely better refactors for each of these, but I'm
optimizing for cleaning up the test output, not for improving the code
interface or possibly introducing breaking changes

bhornseth committed:
Ecosystem Updates

 - Add support for Ruby 3.2 and 3.3
 - Require Faraday 2.0+

bhornseth committed:
Add Verbose option

This change turns off request and response logging by default and allows
users to turn that back on by setting a new `verbose` option to true.

bhornseth committed:
Clean up GitHub Actions script

 - only build on some pushes, but all pull requests
 - support manual builds via workflow_dispatch
 - change publish to be triggered by a GitHub release action
 - update to latest actions/checkout script

wlmcewen commented on CHANGELOG:
```diff
@@ -1,3 +1,10 @@
+v1.7.0
+ - Drop support for ruby 2.7
```
:+1:

wlmcewen reviewed (approved):
Good long overdue changes. Nothing stands out to me, but perhaps the verbose option could be back ported if needed (I'm hoping that's unnecessary).

bhornseth committed:
BREAKING: remove `compress` option

This change removes the `compress` option from the client entirely.

We use the default faraday adapter, which is `net/http` from stdlib. If
you read [the documentation for that module][net-http-docs], you'll come
across this:

> `Net::HTTP` automatically adds Accept-Encoding for compression of
> response bodies and automatically decompresses gzip and deflate
> responses unless a Range header was sent.

In other words, by default, ruby's net/http module will request gzipped
content and automatically uncompress it if that's what the response
contains.

If instead one sets the `Accept-Encoding` header on a request that's
made with `net/http`, that default behavior is ignored and we're left to
deal with unpacking hte response ourselves as was done in the
`FaradayMiddleware` class.

In other words, whether the `compress` option is used or not, the
behavior has been that we're asking for compressed responses and the
option is effectively ignored. I went back to ruby 1.9.2 and the same
behavior existed there. Given that, I'm confident this option has always
been either ignored or redundant.

[net-http-docs]: https://ruby-doc.org/stdlib-3.0.0/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-Compression

bhornseth committed:
Remove highline dependency

Highline is only used within this cli provider, and only to capture and
do some minor formatting on the user input. This changeset removes
highline entirely and replaces it with equivalent ruby

bhornseth committed:
remove broken Code Climate badge

Commit:
Update reso_faraday_middleware.rb

The new Faraday::Middleware class does not have a `body` method from the response and requires us to retrieve it from the `env` variable.
